lexibank_lieberherrkhobwa.py

```diff
@@ -67,36 +92,38 @@ def cmd_install(self, **kw):
 
             # iterate over the source adding lexemes and collecting cognates
             for row in data[2:]:
+                # Get the language_id, which also allows to skip over
+                # data we don't want
                 lid = slug(row[0])
+
                 if lid in langs:
                     for cid in range(1, len(row), 2):
-                        form = row[cid]
-                        if not form or (form == "NA"):
+                        # Extract the value from the raw data, skipping over
+                        # missing or non-existing forms
+                        value = row[cid].strip()
+                        if not value or value == "NA":
                             continue
-                        ipa = form
-                        ipa = ipa.replace("- ", "-").replace(" ", "_")
-                        ipa = strip_brackets(ipa).strip()
 
-                        if ipa:
-                            # split ipa and remove final/initial underscores if any
-                            # (without regular expressions)
-                            ipa = split_text(ipa, "~/~⪤,")[0]
+                        # Build the form, removing spaces between words
+                        # in the same value and stripping brackets
+                        form = value.replace("- ", "-").replace(" ", "_")
```

Comment:
The check for existence is also handled not here, but should be handled by add_values directly, where clean_form is called, and empty rows, defined by empty symbols are ignored.

Reply:
Sorry I don't follow you, also because the diff was not very smart in this snippet. You mean you want to `add_lexemes` to take care of checking for existence?

The existence of the value (or equality to "NA") is checked before building the form. The building of the form is checked after stripping brackets. I tried the keep the logic of the original code as much as possible, I would prefer here to collect the lexemes in one loop and add them in a different one.

Reply:
But this is not the standard we use in pylexibank. The check for existence is also done by pylexibank. Although I still want to change pylexibank to allow for adding forms explicitly, this part is quite crucial: we try to generally handle empty forms and split values. No need for custom operations before, just setting some parameters by overriding the two major functions.

Reply:
The issue here is that the nfilter in `clldutils.text.split_text` will throw an exception if passed an empty string, making checking for existence necessary in some way (not necessarily this `if`).

I will leave the check only for `split_text`, for the time being.

Reply:
Ok, found a way to solve it with more elegant code.